src/delay.js

```diff
@@ -261,52 +260,11 @@ define(function (require) {
     }
   };
 
-  /**
-   *  Set the output level of the delay effect.
-   *  
-   *  @method  amp
-   *  @param  {Number} volume amplitude between 0 and 1.0
-   *  @param {Number} [rampTime] create a fade that lasts rampTime 
-   *  @param {Number} [timeFromNow] schedule this event to happen
-   *                                seconds from now
-   */
-  p5.Delay.prototype.amp = function(vol, rampTime, tFromNow){
-    var rampTime = rampTime || 0;
-    var tFromNow = tFromNow || 0;
-    var now = p5sound.audiocontext.currentTime;
-    var currentVol = this.output.gain.value;
-    this.output.gain.cancelScheduledValues(now);
-    this.output.gain.linearRampToValueAtTime(currentVol, now + tFromNow + .001);
-    this.output.gain.linearRampToValueAtTime(vol, now + tFromNow + rampTime + .001);
-  };
-
-  /**
-   *  Send output to a p5.sound or web audio object
-   *  
-   *  @method  connect
-   *  @param  {Object} unit
-   */
-  p5.Delay.prototype.connect = function(unit) {
-    var u = unit || p5.soundOut.input;
-    this.output.connect(u);
-  };
-
-  /**
-   *  Disconnect all output.
-   *  
-   *  @method disconnect
-   */
-  p5.Delay.prototype.disconnect = function() {
-    this.output.disconnect();
-  };
 
   p5.Delay.prototype.dispose = function() {
-    // remove reference from soundArray
-    var index = p5sound.soundArray.indexOf(this);
-    p5sound.soundArray.splice(index, 1);
 
-    this.input.disconnect();
-    this.output.disconnect();
+	 Effect.prototype.dispose.apply(this);
+
     this._split.disconnect();
     this._leftFilter.disconnect();
     this._rightFilter.disconnect();
```

Comment:
these are p5.Filters, so we should call dispose instead of disconnect to dispose of all its resources/references
```
this._leftFilter.dispose();
this._rightFilter.dispose();
```